src/modules/royalty.ts

```diff
@@ -0,0 +1,152 @@
+import { Royalty, Royalty__factory } from "@3rdweb/contracts";
+import { BigNumberish } from "ethers";
+import { ModuleType, Role, RolesMap } from "../common";
+import { Currency, getCurrencyMetadata } from "../common/currency";
+import { Module } from "../core/module";
+
+interface IRoyalty {
+  /**
+   * Get the total shares held by payees
+   */
+  totalShares(): Promise<BigNumberish>;
+
+  /**
+   * Get the total amount of royalty already released
+   */
+  totalReleased(): Promise<BigNumberish>;
+
+  /**
+   * Returns the amount of `token` released to payees
+   *
+   * @param tokenAddress - The ERC20 token address to check royalty distributions for
+   */
+  totalReleasedByToken(tokenAddress: string): Promise<BigNumberish>;
+
+  /**
+   * Gets the amount of shares held by an account at the address `address`
+   *
+   * @param address - The address to check the shares held for
+   */
+  shares(address: string): Promise<BigNumberish>;
+
+  /**
+   * Gets the amount of royalties already released to the address `address`
+   *
+   * @param address - The address to check the amount of released royalties
+   */
+  released(address: string): Promise<BigNumberish>;
+
+  /**
+   * Gets the amount of royalties already released to the address `address`
+   * for the specific token `tokenAddress`
+   *
+   * @param address - The address to check the amount of released royalties
+   * @param tokenAddress - The ERC20 token address to check royalty distributions for
+   */
+  releasedByToken(address: string, tokenAddress: string): Promise<BigNumberish>;
+
+  /**
+   * Gets the address of the payee at index `index`
+   *
+   * @param index - The index of the payee to get the address of
+   */
+  payee(index: BigNumberish): Promise<string>;
+
+  /**
+   * Triggers the release of the amount of tokens owed to `address` according
+   * to their percentage of the total shares and their previous withdrawals.
+   *
+   * @param address - The address of the payee to release the tokens for
+   */
+  release(address: string): Promise<void>;
```

Comment:
i think currently, most write methods return TransactionReceipt if void.
for a reason that people can keep track of the hashes offchain or do further processing.

in general, i think we need to have a better system in the SDK for TransactionReceipt, i feel like we should almost have a `.on('transactionSent', (transactionReceipt) => void)` event in the SDK so people can subscribe for these tx receipt. (for future notes)